admission-webhook/rockcraft.yaml

```diff
@@ -28,24 +28,23 @@ parts:
       (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
 
   admission-webhook:
-    plugin: go
+    plugin: nil
     build-snaps:
-      - go/1.17/stable
+      - go/1.21/stable
     source: https://github.com/kubeflow/kubeflow.git
-    source-tag: v1.7-branch 
+    source-tag: v1.9.0
+    source-depth: 1
+    source-subdir: components/admission-webhook
     build-environment:
       - CGO_ENABLED: 0
       - GOOS: linux
     override-build: |
```

Comment:
Can we 
1. Switch to using the `go` plugin
2. fully omit the `override-build`
3. Have the go plugin build the file?

Reply:
`go` plugin is not working for me. When I added tests, find out that it's not building binary

Reply:
@TakoB222 the following yaml is working for me

```yaml
name: admission-webhook
base: ubuntu@20.04
version: v1.9.0
summary: An image for Kubeflow's admission-webhook
description: |
  Admission webhook controller in general, intercepts requests to the Kubernetes API server,
  and can modify and/or validate the requests. Here the admission webhook is implemented to
  modify pods based on the available PodDefaults. When a pod creation request is received,
  the admission webhook looks up the available PodDefaults which match the pod's label.
  It then, mutates the Pod spec according to PodDefault's spec.
license: Apache-2.0

platforms:
  amd64:

services:
  base-admission-webhook:
    override: merge
    command: "/webhook"
    startup: enabled
    user: ubuntu

parts:
  admission-webhook:
    plugin: go
    build-snaps:
      - go/1.21/stable
    source: https://github.com/kubeflow/kubeflow.git
    source-tag: v1.9.0
    source-depth: 1
    source-subdir: components/admission-webhook
```

Reply:
@TakoB222 can you also include the previous sections for the ubuntu user? The above was more of a PoC to confirm the go plugin works

Reply:
@kimwnasptd Yes, commit with fix that you suggested already include sections for the ubuntu user